tests/integration/backends/test_tpp.py

```diff
@@ -645,6 +648,7 @@ def test_ons_cis(select_all):
         {
             "patient_id": 1,
             "visit_date": date(2021, 10, 20),
+            "visit_id": base64.b16encode(hashlib.sha256(b"1").digest()).decode(),
```

Comment:
I think you can just do:
```suggestion
            "visit_id": hashlib.sha256(b"1").digest().hex(),
```

In fact, you could use the `.hexdigest()` method, but that's specific to `hashlib` whereas `.hex()` is a method on all bytestrings so I think it's preferable here.

Reply:
Thanks, Dave. Both `.hexdigest()` and `.hex()` seem to use characters `[a-z]`, whilst `b16encode.decode()` and T-SQL seem to use characters `[A-Z]`. I've force pushed a change that calls `.upper()` on the resulting hex.

https://github.com/opensafely-core/ehrql/blob/0bfd64f0be5cb15e6b4390d2e8058f0060973a7c/tests/integration/backends/test_tpp.py#L651

Reply:
Ah, makes sense. Thanks.